invenio_rdm_records/ext.py

```diff
@@ -112,7 +86,6 @@ def init_app(self, app):
         self.init_config(app)
         self.init_services(app)
         self.init_resource(app)
-        app.before_request(verify_token)
```

Comment:
Making changes to the identity from the `before_request` here and from the `on_identity_loaded` signal handler can cause issues because the order of execution is not controlled. The way to avoid this is to move identity logic to 1 place --> inside the signal handler `on_identity_loaded` (see above)